src/main/java/net/countercraft/movecraft/combat/features/FlammableOverride.java

```diff
@@ -0,0 +1,186 @@
+                        }
+                    } catch (NumberFormatException | NullPointerException ex) {
+                        MovecraftCombat.getInstance().getLogger()
+                                .warning("Unable to load " + m.name() + ": " + entry.getValue());
+                        continue;
+                    }
+                    flammabilityOverride.put(m, new Pair<>(flammability, encouragement));
+                }
+            }
+
+            String[] parts = Bukkit.getServer().getMinecraftVersion().split("\\.");
+            if (parts.length < 2)
+                throw new IllegalArgumentException();
+            int major_version = Integer.parseInt(parts[1]);
+            if (major_version < 20) {
+                nmsInterface = new FlammabilityNMS_SpigotMappings(); // Tested on 1.18.2 and 1.19.4
+            } else {
+                nmsInterface = new FlammabilityNMS_MojangMappings(); // Tested on 1.20.6
+            }
+        } catch (Exception e) {
+            MovecraftCombat.getInstance().getLogger().info("Failed to load FlammabilityOverride: ");
+            e.printStackTrace();
+        }
+    }
+
+    public static boolean setFlammability(Material m, int flammability, int encouragement) {
+        return nmsInterface.setBlastResistance(m, flammability, encouragement);
+    }
+
+    public static boolean revertToVanilla(Material m) {
+        return true;
+        //return setFlammability(m, m.getBlastResistance()); // Spigot stores the vanilla values for us
+    }
+
+    public static void enable() {
+        try {
+            for (var entry : flammabilityOverride.entrySet()) {
+                if (!setFlammability(entry.getKey(), entry.getValue().getLeft(), entry.getValue().getRight()))
+                    MovecraftCombat.getInstance().getLogger()
+                            .warning("Unable to set " + entry.getKey().name() + ": " + entry.getValue());
+            }
+        } catch (Exception e) {
+            MovecraftCombat.getInstance().getLogger().info("Failed to enable FlammabilityOverride: ");
+            e.printStackTrace();
+        }
+    }
+
+    public static void disable() {
+        try {
+            for (Material m : flammabilityOverride.keySet()) {
+                if (!revertToVanilla(m))
+                    MovecraftCombat.getInstance().getLogger().warning("Unable to reset " + m.name());
+            }
+        } catch (Exception e) {
+            MovecraftCombat.getInstance().getLogger().info("Failed to disable FlammabilityOverride: ");
+            e.printStackTrace();
+        }
+    }
+
+    private static class FlammabilityNMS {
+
+        public boolean setBlastResistance(Material m, int flammability, int encouragement) {
+            throw new NotImplementedException();
+        }
+
+        protected static void writeField(@NotNull Object fireBlock, @NotNull Object block, String igniteOddsName, String burnOddsName, int flammability, int encouragement)
```

Comment:
Instead of passing in both fields here, the function could be made generic to handle them in separate calls.

Reply:
Good idea, i'll do that tomorrow or maybe later.